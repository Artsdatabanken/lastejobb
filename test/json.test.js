const { json } = require("..");

const input = {
  z: 2,
  k: [6, 1, 3],
  y: { c: 5, a: "abc", inception: new Date(1880, 1, 2) }
};
const inputArray = [6, 1, 3];

// Recursively sort objects
test("sort JSON object", () => {
  const actual = json.sortKeys(input);
  expect(actual).toMatchSnapshot();
});

// We don't sort the array
test("sort JSON array", () => {
  const actual = json.sortKeys(inputArray);
  expect(actual).toMatchSnapshot();
});

test("arrayToObject", () => {
  const arr = [{ id: 5, value: "abc" }, { id: 2, name: "def" }];
  const actual = json.arrayToObject(arr, { uniqueKey: "id" });
  expect(actual).toMatchSnapshot();
});

test("arrayToObject: keep keys", () => {
  const arr = [{ id: 5, value: "abc" }, { id: 2, name: "def" }];
  const actual = json.arrayToObject(arr, {
    uniqueKey: "id",
    removeKeyProperty: false
  });
  expect(actual).toMatchSnapshot();
});

test("objectToArray", () => {
  const o = { a: { id: 5 }, b: { name: "def" } };
  const actual = json.objectToArray(o, "identifier");
  expect(actual).toMatchSnapshot();
});

test("moveKey", () => {
  const o = { a: { id: 5 }, b: { c: { name: "def" } } };
  const actual = json.moveKey(o, "a", "b.c.a");
  expect(actual).toMatchSnapshot();
});

test("remove empty keys", () => {
  const input = { a: { b: {} }, c: 2 };
  const actual = json.removeEmptyKeys(input);
  expect(actual).toMatchSnapshot();
});

test("merge deep", () => {
  const obj1 = {
    a: 1,
    b: 1,
    c: { x: 1, y: 1 },
    d: [1, 1]
  };
  const obj2 = {
    b: 2,
    c: { y: 2, z: 2 },
    d: [2, 2],
    e: 2
  };
  const actual = json.mergeDeep(obj1, obj2);
  expect(actual).toMatchSnapshot();
});
