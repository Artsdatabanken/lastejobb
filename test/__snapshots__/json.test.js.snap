// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrayToObject 1`] = `
Object {
  "2": Object {
    "name": "def",
  },
  "5": Object {
    "value": "abc",
  },
}
`;

exports[`arrayToObject: keep keys 1`] = `
Object {
  "2": Object {
    "id": 2,
    "name": "def",
  },
  "5": Object {
    "id": 5,
    "value": "abc",
  },
}
`;

exports[`merge deep 1`] = `
Object {
  "a": 1,
  "b": 2,
  "c": Object {
    "x": 1,
    "y": 2,
    "z": 2,
  },
  "d": Array [
    1,
    1,
    2,
    2,
  ],
  "e": 2,
}
`;

exports[`moveKey 1`] = `
Object {
  "b": Object {
    "c": Object {
      "a": Object {
        "id": 5,
      },
      "name": "def",
    },
  },
}
`;

exports[`objectToArray 1`] = `
Array [
  Object {
    "id": 5,
    "identifier": "a",
  },
  Object {
    "identifier": "b",
    "name": "def",
  },
]
`;

exports[`remove empty keys 1`] = `
Object {
  "c": 2,
}
`;

exports[`sort JSON array 1`] = `
Array [
  6,
  1,
  3,
]
`;

exports[`sort JSON object 1`] = `
Object {
  "k": Array [
    6,
    1,
    3,
  ],
  "y": Object {
    "a": "abc",
    "c": 5,
    "inception": 1880-02-01T23:17:00.000Z,
  },
  "z": 2,
}
`;
